provider/mem/mem.go

```diff
@@ -178,8 +194,11 @@ func (a *Alerts) GetPending() provider.AlertIterator {
 
 	go func() {
 		defer close(ch)
+		a.mtx.Lock()
```

Comment:
Since we are locking a lot more now, should we consider using `RWMutex` for the lock? concurrent `GetPending` and `count` should be safe right?